File: Data Structures & Algorithms/string-encode-and-decode/submission-0.cpp
class Solution {
public:

    string encode(vector<string>& s) {
        if(s.size() == 0) return "";
        string ans = "|";
        for(int i=0; i<s.size(); i++) ans+=(s[i]+"|");
        return ans;
    }

    vector<string> decode(string s) {
        vector<string> ans;
        cout<<s<<endl;
        if(s=="") return ans;
        int p=0;
        for(int i=1; i<s.length(); i++){
            if(s[i]=='|'){
                string x = "";
                for(int j=p+1; j<i;j++) x+=s[j];
                ans.push_back(x);
                p=i;
            }
        }

        return ans;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    private int ans;

    private void isGood(TreeNode root, int routeMax) {
        if (root == null) {
            return;
        }

        if (routeMax <= root.val) {
            ans++;
        }

        isGood(root.left, Math.max(routeMax, root.val));
        isGood(root.right, Math.max(routeMax, root.val));
    }

    public int goodNodes(TreeNode root) {
        ans = 0;
        isGood(root, Integer.MIN_VALUE);
        return ans;
    }
}
